Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -379,38 +332,36 @@ void CGPathAddArc(CGMutablePathRef path,
                   bool clockwise) {
     RETURN_IF(!path);
 
-    // Normalize the start angle so it's between 0 and 2*pi
-    startAngle = fmod(startAngle, 2.0f * M_PI);
-    if (startAngle < 0.0f) {
-        startAngle += 2.0f * M_PI;
-    }
+    CGPoint startPoint = __CreateCGPointWithTransform(x + radius * cos(startAngle), y + radius * sin(startAngle), transform);
+    CGPoint endPoint = __CreateCGPointWithTransform(x + radius * cos(endAngle), y + radius * sin(endAngle), transform);
 
-    // Normalize the end angle so it's between 0 and 2*pi
-    endAngle = fmod(endAngle, 2.0f * M_PI);
-    if (endAngle < 0.0f) {
-        endAngle += 2.0f * M_PI;
+    if (!CGPointEqualToPoint(path->GetCurrentPoint(), startPoint)) {
+        CGPathAddLineToPoint(path, nullptr, startPoint.x, startPoint.y);
     }
 
-    // Calculate the starting point of the arc
-    CGPoint arcStart = CGPointMake((cos(startAngle) * radius) + x, (sin(startAngle) * radius) + y);
-
-    // Either draw a line to or move to the start of the arc
-    if (!CGPathIsEmpty(path)) {
-        CGPathAddLineToPoint(path, transform, arcStart.x, arcStart.y);
+    // Create the parameters for the AddArc method.
+    const D2D1_POINT_2F endPointD2D = _CGPointToD2D_F(endPoint);
+    const D2D1_SIZE_F radiusD2D = { radius, radius };
+    CGFloat rotationAngle = abs(startAngle - endAngle);
+    D2D1_ARC_SIZE arcSize = D2D1_ARC_SIZE_SMALL;
+    CGFloat expectedAngle = (clockwise ? startAngle + rotationAngle : startAngle - rotationAngle);
+
+    // D2D does not understand that the ending angle must be pointing in the proper direction, thus we must translate
+    // what it means to have an ending angle to the proper small arc or large arc that D2D will use since a circle will
+    // intersect that point regardless of which direction it is drawn in.
+    if (expectedAngle == endAngle) {
+        arcSize = D2D1_ARC_SIZE_LARGE;
     } else {
-        CGPathMoveToPoint(path, transform, arcStart.x, arcStart.y);
+        rotationAngle = (2 * M_PI) - rotationAngle;
     }
+    D2D1_SWEEP_DIRECTION sweepDirection = { clockwise ? D2D1_SWEEP_DIRECTION_COUNTER_CLOCKWISE : D2D1_SWEEP_DIRECTION_CLOCKWISE };
```

Comment:
Correct.

---
In reply to: [88087978](https://github.com/Microsoft/WinObjC/pull/1384#discussion_r88087978) [](ancestors = 88087978)